apollo-ios-codegen/Sources/GraphQLCompiler/CompilationResult.swift

```diff
@@ -139,12 +142,19 @@ public final class CompilationResult: JavaScriptObjectDecodable {
       self.variables = variables
       self.rootType = rootType
       self.selectionSet = selectionSet
-      self.directives = directives
       self.referencedFragments = referencedFragments
       self.source = source
       self.filePath = filePath
       self.isLocalCacheMutation = directives?
         .contains { $0.name == Constants.DirectiveNames.LocalCacheMutation } ?? false
+        
+      let referencedImports: [Directive] = self.referencedFragments
```

Comment:
I see what you're doing here, but I don't think we should be adding the referenced fragment's `@import` directives to the parent's `directives`. We should add them to the `moduleImports` but not the actual `directives` property.

Reply:
This means `moduleImports` probably just becomes a stored property instead of being computed.